config/v1/types_apiserver.go

```diff
@@ -191,9 +194,23 @@ type APIServerEncryption struct {
 	// +unionDiscriminator
 	// +optional
 	Type EncryptionType `json:"type,omitempty"`
+
+	// kms defines the configuration for the external KMS instance that manages the encryption keys,
+	// when KMS encryption is enabled sensitive resources will be encrypted using keys managed by an
+	// externally configured KMS instance.
+	//
+	// The Key Management Service (KMS) instance provides symmetric encryption and is responsible for
+	// managing the lifecyle of the encryption keys outside of the control plane.
+	// This allows integration with an external provider to manage the data encryption keys securely.
+	//
+	// +openshift:enable:FeatureGate=KMSv2
+	// +unionMember
+	// +optional
+	KMS *KMSConfig `json:"kms,omitempty"`
```

Comment:
You'll need to add CEL validation to the parent struct to ensure this is only set when type is KMS. Look for CELUnion in the example v1 API group stable_types.go for an example of how to do this

Reply:
yeah, I've added validation to enforce KMSconfig only when type is KMS.